src/votekit/ballot_generator.py

```diff
@@ -1201,4 +1201,144 @@ def __init__(self, **data):
         k = len(data["candidates"])
         # Call the parent class's __init__ method to handle common parameters
         super().__init__(k=k, **data)
-        
\ No newline at end of file
+        
+
+class SlatePreference(BallotGenerator):
+    """
+    Class for generating ballots using a slate preference model. 
+    This model first samples a ballot type by flipping a cohesion parameter weighted coin.
+    It then fills out the ballot type via sampling with out replacement from the interval.
+
+    Only works with 2 blocs at the moment.
+    
+    Can be initialized with an interval or can be
+    constructed with the Dirichlet distribution using the `from_params` method in the
+    `BallotGenerator` class.
+
+    **Attributes**
+
+    `slate_to_candidates`
+    :   dictionary mapping of slate to candidates (ex. {bloc: [candidate]}).
+
+    `pref_interval_by_bloc`
+    :   dictionary mapping of bloc to preference interval.
+        (ex. {bloc: {candidate : interval length}}).
+
+    `bloc_voter_prop`
+    :   dictionary mapping of bloc to voter proportions (ex. {bloc: proportion}).
+
+    `cohesion_parameters`
+    :   dictionary mapping of slate to cohesion level (ex. {bloc: .7}).
+
+    **Methods**
+
+    See `BallotGenerator` base class
+    """
+
+    def __init__(self, slate_to_candidates: dict = {}, cohesion_parameters: dict = {}, **data):
+        # Call the parent class's __init__ method to handle common parameters
+        super().__init__(**data)
+        self.slate_to_candidates = slate_to_candidates
+        self.cohesion_parameters = cohesion_parameters
+    
+    def generate_profile(self, number_of_ballots: int, by_bloc: bool = False) \
+                                                            -> Union[PreferenceProfile, Tuple]:
+        # the number of ballots per bloc is determined by Huntington-Hill apportionment
+        blocs = list(self.bloc_voter_prop.keys())
+        bloc_props = list(self.bloc_voter_prop.values())
```

Comment:
note: since these methods are called a lot, I'm going to point out small speed issues that might accumulate.  feel free to ignore since most are quite small, but they could add up in practice.

these two casts to list probably aren't needed, and if they can be removed will save memory/time

Reply:
It's not super easy to avoid casting them as a list, since the apportionment package needs `bloc_props` as a list. Moreover, these number of blocs is almost always going to be small so the cloning here shouldn't be too expensive.

I think the bigger fix here is to create an attribute `self.blocs` in the init method of all of these ballot generators since it get's used so often, and then at least it's only once per object, not once per call to generator.